Samples/NetworkMediaStreamer/RTSPServer/inc/RtspSession.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+#pragma once
+
+// supported command types
+enum class RTSP_CMD
+{
+    UNKNOWN = 0,
+    OPTIONS,
+    DESCRIBE,
+    SETUP,
+    PLAY,
+    TEARDOWN
+};
+#define RTP_DEFAULT_PORT 54554
+#define RTSP_BUFFER_SIZE       10000    // for incoming requests, and outgoing responses
+#define RTSP_PARAM_STRING_MAX  200
+class RTSPSession
+{
+public:
+    RTSPSession(
+        CSocketWrapper* rtspClientSocket,
+        winrt::Windows::Foundation::Collections::PropertySet streamers,
+        IRTSPAuthProvider* pAuthProvider,
+        winrt::event<winrt::LogHandler>* m_pLoggers);
+    virtual ~RTSPSession();

```

Comment:
>virtual [](start = 4, length = 7)

not needed appears to be a sealed class

Reply:
not sealed

---
In reply to: [491109072](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491109072) [](ancestors = 491109072)

Reply:
there are no interfaces to override and the rest of the class members and methods are marked private, what should they override or extend

---
In reply to: [491111659](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491111659) [](ancestors = 491111659,491109072)

Reply:
HandleRequest is public. One can extend this class and have an extended implementation of RTSPSession to handle commands like "ANNOUNCE" or "RECORD" and use a different/extended implementation of RTSPServer to manage it.

---
In reply to: [491206435](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491206435) [](ancestors = 491206435,491111659,491109072)